src/validators/definitions.rs

```diff
@@ -100,14 +98,60 @@ impl Validator for DefinitionRefValidator {
         }
     }
 
+    fn validate_assignment<'s, 'data: 's>(
+        &'s self,
+        py: Python<'data>,
+        obj: &'data PyAny,
+        field_name: &'data str,
+        field_value: &'data PyAny,
+        extra: &Extra,
+        slots: &'data [CombinedValidator],
+        recursion_guard: &'s mut RecursionGuard,
```

Comment:
This solves a bug where `validate_assignment` straight up didn't work with any def refs because they didn't implement it.